benchmarks/benchmarks.py

```diff
@@ -0,0 +1,33 @@
+# Write the benchmarking functions here.
+# See "Writing benchmarks" in the asv docs for more information.
+from .runner import *
+
+class TimeSuite:
+    """
+    An example benchmark that times the performance of various kinds
+    of iterating over dictionaries in Python.
+    """
+    def setup(self, nworkers =1):
+        self.ports = random_ports(self.nworkers + 1)
```

Comment:
As a general rule, these variables only need to be attributes (`self.`) if they are going to be used *outside* this function (in teardown or timer functions).

For example, `nworkers` is an input argument, but there is no `self.nworkers` defined. Since we only use it here, we can use just `nworkers` instead of `self.nworkers`